programs/survival/silk_budding_amethyst.sc

```diff
@@ -0,0 +1,31 @@
+// Conversion of Silk Spwaners; Implemented by Scott Gudeman (DragonCrafted87)
+// See https://github.com/gnembon/scarpet/blob/master/programs/survival/silk_spawners.sc for Original Basis
+
+//keeps script loaded upon server start
+__config() -> {'stay_loaded' -> true, 'scope' -> 'global'};
+
+// private __silk_budding_amethyst (player, block)
+// returns: true if and only if
+//  - player is holding a silk touch diamond or netherite pickaxe
+//  - block is a budding_amethyst
+__silk_budding_amethyst (player, block) -> (
+        if (block != 'budding_amethyst', return());
+        tool = player ~ 'holds';
+        if ((!tool || (get(tool, 0) != 'diamond_pickaxe' && get(tool, 0) != 'netherite_pickaxe')), return());
+        nbt = get(tool, 2);
+        if (!nbt, return());
+        ench = get(nbt, 'Enchantments[]');
+        if (!ench, return());
+        if (type(ench) != 'list', ench = l(ench));
+        for (ench, if (get(_, 'id') == 'minecraft:silk_touch', return(true)));
+        false
```

Comment:
Try to avoid using unnecessary `return()` calls, instead put it all into an if statement, checking if it's a budding amethyst block, and if the player is holding a netherite or diamond pick with silk touch. If you get creative (not that you have to though), you could squeeze it all into a single line, though that may be slow and hard to read/understand

Reply:
The reason for fewer return calls is cos they're kinda slow for the scarpet interpreter, so doing them very often is slow.

Reply:
wound up basically rewriting the whole function and was actually able to simplify it a bit with some of the examples i found in the api :)